### PR TITLE
Add Microsoft.DotNet.XliffTasks into arcade

### PR DESCRIPTION
**Pull request by ViktorHofer**
Merge https://github.com/dotnet/xliff-tasks into arcade. The repository only consists of two non-shipping projects:
- Microsoft.DotNet.XliffTasks and
- Microsoft.DotNet.XliffTasks.Tests

By merging this repository into Arcade, we can reduce the build graph and make VMR and source-build more efficient. In addition, the cost of maintaining xliff-tasks's infrastructure was way too high in contrast to its benefit (single project). By moving the project into Arcade, it follows the branding and branching model from Arcade which is general goodness over the single branch in xliff-tasks.

This brings over the commits from the old repository and adds the projects to Arcade's solution file.

Post-merge actions:
1. Disable the AzDO pipeline for xliff-tasks
2. After an official build succeeded with these changes, update the Version.Details.xml entry in Arcade.
3. Remove the VMR project and dependencies pointing to it: https://github.com/dotnet/dotnet/blob/main/repo-projects/xliff-tasks.proj
4. Update the xliff-tasks entry in all consuming repository's Version.Details.xml files to point to arcade instead of xliff-tasks.
5. Delete all subscriptions from or to xliff-tasks
6. Archive the xliff-tasks repository. Migrate the remaining issue over into Arcade.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by ViktorHofer** [comment]
cc @MichaelSimons @mthalman

**Comment by ViktorHofer** [comment]
@markwilkie approved the merge via mail

File: Directory.Packages.props [hunk 1]
```diff
@@ -18,6 +18,7 @@
   <ItemGroup>
     <!-- arcade -->
     <PackageVersion Include="Microsoft.DotNet.SwaggerGenerator.MSBuild" Version="$(MicrosoftDotNetSwaggerGeneratorMSBuildVersion)" />
+    <PackageVersion Include="Microsoft.DotNet.XliffTasks" Version="$(MicrosoftDotNetXliffTasksVersion)" />
```
Comment (mmitche) [inline Directory.Packages.props:21]: Is this necessary? It looks like there are two package refs in arcade:
- src/Microsoft.DotNet.AsmDiff/Microsoft.DotNet.AsmDiff.csproj - I think this should change to a ProjectRef
- src/Microsoft.DotNet.Arcade.Sdk/tools/Localization.targets - This is packaged with the arcade SDK and is just a version reference.

File: eng/Versions.props [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <MicrosoftDotNetBuildTasksFeedVersion>9.0.0-beta.23518.2</MicrosoftDotNetBuildTasksFeedVersion>
     <MicrosoftDotNetSignToolVersion>9.0.0-beta.23518.2</MicrosoftDotNetSignToolVersion>
     <MicrosoftDotNetSwaggerGeneratorMSBuildVersion>9.0.0-beta.23518.2</MicrosoftDotNetSwaggerGeneratorMSBuildVersion>
+    <MicrosoftDotNetXliffTasksVersion>1.0.0-beta.23523.1</MicrosoftDotNetXliffTasksVersion>
```
Comment (mmitche) [inline eng/Versions.props:16]: I think this can be removed?

File: Directory.Packages.props (hunk 1 above)
Comment (ViktorHofer) [reply Directory.Packages.props]: Unfortunately using `ProjectReference` doesn't make this easily possible, see https://github.com/NuGet/Home/issues/6624.

That said, as we just recently had to solve this in runtime for the live linker, I was able to use most of that infrastructure and make it work. Please see my latest commit. Tried this out locally, changing the resx causes the xlfs to be updated during the build.

File: eng/Microsoft.DotNet.XliffTasks.InTree.targets [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<Project>
```
Comment (mmitche) [inline eng/Microsoft.DotNet.XliffTasks.InTree.targets:1]: Can you add a comment here as to why this is necessary (underlying issue)

**Review by mmitche — APPROVED** [review]

**Review by premun — APPROVED** [review]

**Comment by MichaelSimons** [comment]
CC @dotnet/source-build-internal for visibility.

**Comment by mthalman** [comment]
@ViktorHofer - This will require the removal of https://github.com/dotnet/installer/blob/main/src/SourceBuild/content/repo-projects/xliff-tasks.proj and its reference from https://github.com/dotnet/installer/blob/fb7c9717cfd661b120e900a073c60a8fc9e898da/src/SourceBuild/content/repo-projects/dotnet.proj#L29.

That PR can be prepared now but won't succeed until the new Arcade flows into installer.

**Comment by ViktorHofer** [comment]
Thanks for the notice. Yes I discussed that with Michael offline couple of days ago. I will add the required changes into the next arcade --> installer dependency update PR.

**Pull request merged**